### PR TITLE
Newest Changes

### PR DESCRIPTION
**Pull request by juanmc2005**
## Changelog
- Custom seed, margin and thershold are now script parameters
- SemEval modes are now augmentation objects
- Separated SemEval evaluation experiments to cover logit and embedding based evaluations
- t-SNE visualization is possible in STS, while still tracking the original phrases for each embedding
- SemEval evaluators refactored into class hierarchy
- Experiments have been splitted into different files
- Added customized experience id for each script launch, which changes the name of the output models and logs
- Mean loss for each epoch is now being logged to a file too
- Added RegularModelSaver to save models every x epochs
- Removed mutable default parameters
- Added STS forward mode objects to change the way sentences are fed to the model (single, pair, triplets)
- Bugfixes and minor improvements

### CODE REVIEW

File: main.py [hunk 1]
```diff
@@ -30,38 +36,39 @@
 parser.set_defaults(save=True)
 parser.add_argument('--task', type=str, default='mnist', help='The task to train')
 parser.add_argument('--recover', type=str, default=None, help='The path to the saved model to recover for training')
+parser.add_argument('--margin', type=float, default=2., help='The margin to use for the losses that need it')
```
Comment (juanmc2005) [inline main.py:39]: Could also give the option '-m'

File: main.py [hunk 2]
```diff
@@ -30,38 +36,39 @@
...
+parser.add_argument('--t', type=float, default=3., help='The threshold for STS to consider a pair positive or negative')
```
Comment (juanmc2005) [inline main.py:40]: Should be either '--threshold' or '-t'
